components/TreasuryAccount/ConvertToStSol.tsx

```diff
@@ -0,0 +1,293 @@
+        {stSolTokenAccounts.length > 0 && (
+          <GovernedAccountSelect
+            label="stSOL Treasury account"
+            governedAccounts={stSolTokenAccounts as AssetAccount[]}
+            shouldBeGoverned={false}
+            governance={currentAccount?.governance}
+            value={form.destinationAccount}
+            onChange={(evt) =>
+              handleSetForm({
+                value: evt,
+                propertyName: 'destinationAccount',
+              })
+            }
+            error={formErrors['destinationAccount']}
+            noMaxWidth={true}
+          ></GovernedAccountSelect>
+        )}
+        <Input
+          min={mintMinAmount}
+          label="Amount SOL"
+          type="number"
+          value={form.amount}
+          step={mintMinAmount}
+          onChange={(evt) =>
+            handleSetForm({
+              value: evt.target.value,
+              propertyName: 'amount',
+            })
+          }
+          onBlur={(evt) =>
+            handleSetForm({
+              value: parseFloat(
+                Math.max(
+                  Number(mintMinAmount),
+                  Math.min(
+                    Number(Number.MAX_SAFE_INTEGER),
+                    Number(evt.target.value)
+                  )
+                ).toFixed(precision(mintMinAmount))
+              ),
+              propertyName: 'amount',
+            })
+          }
+          error={formErrors['amount']}
+          noMaxWidth={true}
+        />
+        <div
+          className="flex items-center hover:cursor-pointer w-24"
+          onClick={() => setShowOptions(!showOptions)}
+        >
+          <div className="h-4 w-4 mr-1 text-primary-light">
+            {showOptions ? <ArrowCircleUpIcon /> : <ArrowCircleDownIcon />}
+          </div>
+          <small className="text-fgd-3">Options</small>
+        </div>
+        {showOptions && (
+          <>
+            <Input
+              noMaxWidth={true}
+              label="Title"
+              value={form.title}
+              type="text"
+              placeholder={
+                form.amount && form.destinationAccount
+                  ? getProposalText(form.amount)
+                  : 'Title of your proposal'
+              }
+              onChange={(evt) =>
+                handleSetForm({
+                  value: evt.target.value,
+                  propertyName: 'title',
+                })
+              }
+            />
+            <Textarea
+              noMaxWidth={true}
+              label="Description"
+              placeholder={
+                'Description of your proposal or use a github gist link (optional)'
+              }
+              wrapperClassName="mb-5"
+              value={form.description}
+              onChange={(evt) =>
+                handleSetForm({
+                  value: evt.target.value,
+                  propertyName: 'description',
+                })
+              }
+            ></Textarea>
```

Comment:
small style nit, if you could use a self-closing jsx tag instead of `</Textarea>` that would be appreciated